### PR TITLE
Refactor manifest file handling in cli (#3186)

### PR DESCRIPTION
**Pull request by sbbroot**
* Create the ManifestReader class for handling configuraton data used by
  the epicli

* Rename all of the manifest variables to input/output_manifest

### CODE REVIEW

File: cli/src/ansible/AnsibleVarsGenerator.py [hunk 1]
```diff
@@ -120,11 +120,11 @@ def write_role_vars(self, ansible_dir, role, document, vars_file_name='main.yml'
 
     def write_role_manifest_vars(self, ansible_dir, role, kind):
         try:
-            cluster_model = select_single(self.manifest_docs, lambda x: x.kind == 'epiphany-cluster')
-        except ExpectedSingleResultException:
+            cluster_model = self.mhandler['epiphany-cluster']
+        except IndexError:
```
Comment (tomasz-baran) [inline cli/src/ansible/AnsibleVarsGenerator.py:120]: Can we have a manifest without `epiphany-cluster` kind here? If not, IMO it's better to not "hide" the exception.

**Review by seriva — COMMENTED** [review]
Is there any documentation that needs to have the update from --file > --input-manifest ?

**Comment by sbbroot** [comment]
> Is there any documentation that needs to have the update from --file > --input-manifest ?

I didn't find anything related to the `--file` flag in our documentation.

**Review by seriva — DISMISSED** [review]

**Review by tomasz-baran — DISMISSED** [review]

**Review by tomasz-baran — DISMISSED** [review]

**Review by seriva — DISMISSED** [review]

**Review by seriva — DISMISSED** [review]

File: cli/epicli.py [hunk 2]
```diff
@@ -195,7 +195,7 @@ def apply_parser(subparsers):
     required = sub_parser.add_argument_group('required arguments')
 
     #required
-    required.add_argument('-f', '--input-manifest', dest='input_manifest', type=Path, required=True,
+    required.add_argument('-f', '--input-manifest', dest='input_manifest', type=str, required=True,
```
Comment (tomasz-baran) [inline cli/epicli.py:196]: After this change import of Path (line 11) is useless.

**Review by tomasz-baran — CHANGES_REQUESTED** [review]

**Review by seriva — DISMISSED** [review]

**Review by tomasz-baran — DISMISSED** [review]

**Review by seriva — DISMISSED** [review]

**Review by przemyslavic — APPROVED** [review]

**Review by seriva — APPROVED** [review]

**Pull request merged**